packages/expo-intent-launcher/android/src/main/java/expo/modules/intentlauncher/IntentLauncherModule.java

```diff
@@ -102,7 +102,14 @@ public void startActivity(String activityAction, @NonNull ReadableArguments para
 
     mUIManager.registerActivityEventListener(this);
     mPendingPromise = promise;
-    activity.startActivityForResult(intent, REQUEST_CODE);
+
+    try {
+      activity.startActivityForResult(intent, REQUEST_CODE);
+    }
+    catch (ActivityNotFoundException e) {
+      mPendingPromise.reject(e);
```

Comment:
I wonder what's the error message here? 🤔 
Do you mind providing one, based on your experience?
We tend to suggest what's wrong when some error is thrown, so I guess we should wrap this `ActivityNotFoundException` and provide some more info about possible solution.

I guess it would might look like this:
- create another exception class `NoActivityForIntentException` in this directory: https://github.com/expo/expo/tree/master/packages/expo-intent-launcher/android/src/main/java/expo/modules/intentlauncher/exceptions
```java
package expo.modules.intentlauncher.exceptions;

public class NoActivityForIntentException extends CodedException implements CodedThrowable {
  public ActivityAlreadyStartedException(String activityAction, ActivityNotFoundException e) {
    super("Failed to find activity for " + activityAction + ". Ensure that there's an application on your device capable of opening such intent or if the intent has correct format.", e);
  }

  @Override
  public String getCode() {
    return "E_NO_ACTIVITY_FOR_INTENT";
  }
}
```
- and change this code fragment like this:
```suggestion
      mPendingPromise.reject(new NoActivityForIntentException(activityAction, e));
```

cc @cruzach, @brentvatne regarding suggested error message

Reply:
![Screenshot_1614954788](https://user-images.githubusercontent.com/22592111/110129334-cd0abf00-7e02-11eb-8574-c7504b4ae862.png)

This is the promise rejection warning that gets thrown. I think it is already pretty clear.

Reply:
i'm ok with this error message, what do you think @bbarthec?

Reply:
Looks good to me 👍 
Let it be merged then! 🎉